test/unit/lib/commands/drivers/package-utils.test.ts

```diff
@@ -0,0 +1,385 @@
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/init.lua')
+			expect(zipFileMock).toHaveBeenCalledTimes(1)
+			expect(zipFileMock).toHaveBeenCalledWith('src/init.lua', readStreamMock)
+		})
+
+		it('includes nested files', () => {
+			requireDirSpy.mockReturnValueOnce('src dir')
+			isFileSpy.mockReturnValueOnce(true) // init.lua exists
+			readdirSyncMock.mockReturnValueOnce(['init.lua', 'subdirectory'])
+			isDirSpy.mockReturnValueOnce(false) // init.lua is not a directory
+			isDirSpy.mockReturnValueOnce(true) // subdirectory is a directory
+			readdirSyncMock.mockReturnValueOnce(['lib.lua'])
+			isDirSpy.mockReturnValueOnce(false) // lib.lua is not directory
+
+			expect(() => processSrcDir('project dir', zipMock, [])).not.toThrow()
+
+			expect(requireDirSpy).toHaveBeenCalledTimes(1)
+			expect(requireDirSpy).toHaveBeenCalledWith('project dir/src')
+			expect(isFileSpy).toHaveBeenCalledTimes(1)
+			expect(isFileSpy).toHaveBeenCalledWith('src dir/init.lua')
+			expect(readdirSyncMock).toHaveBeenCalledTimes(2)
+			expect(readdirSyncMock).toHaveBeenCalledWith('src dir')
+			expect(readdirSyncMock).toHaveBeenCalledWith('src dir/subdirectory')
+			expect(isDirSpy).toHaveBeenCalledTimes(3)
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/init.lua')
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/subdirectory')
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/subdirectory/lib.lua')
+			expect(zipFileMock).toHaveBeenCalledTimes(2)
+			expect(zipFileMock).toHaveBeenCalledWith('src/init.lua', readStreamMock)
+			expect(zipFileMock).toHaveBeenCalledWith('src/subdirectory/lib.lua', readStreamMock)
+		})
+
+		it('skips files that match test dir pattern', () => {
+			requireDirSpy.mockReturnValueOnce('src dir')
+			isFileSpy.mockReturnValueOnce(true) // init.lua exists
+			readdirSyncMock.mockReturnValueOnce(['init.lua', 'test.lua'])
+			isDirSpy.mockReturnValueOnce(false) // init.lua is not a directory
+			isDirSpy.mockReturnValueOnce(false) // test.lua is not a directory
+			const matcher = jest.fn()
+				.mockReturnValueOnce(false) // init.lua is not a test file
+				.mockReturnValueOnce(true) // test.lua is a test file
+
+			expect(() => processSrcDir('project dir', zipMock, [matcher])).not.toThrow()
+
+			expect(requireDirSpy).toHaveBeenCalledTimes(1)
+			expect(requireDirSpy).toHaveBeenCalledWith('project dir/src')
+			expect(isFileSpy).toHaveBeenCalledTimes(1)
+			expect(isFileSpy).toHaveBeenCalledWith('src dir/init.lua')
+			expect(readdirSyncMock).toHaveBeenCalledTimes(1)
+			expect(readdirSyncMock).toHaveBeenCalledWith('src dir')
+			expect(isDirSpy).toHaveBeenCalledTimes(2)
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/init.lua')
+			expect(isDirSpy).toHaveBeenCalledWith('src dir/test.lua')
+			expect(zipFileMock).toHaveBeenCalledTimes(1) // NOT 2! :-)
+			expect(zipFileMock).toHaveBeenCalledWith('src/init.lua', readStreamMock)
+		})
+
+		it('throws error if nesting is too deep', () => {
```

Comment:
It would be helpful to have a comment here pointing out that nesting starts at 2 in `processSrcDir`. It was confusing to see limit of 10 but then 2 for loops 1<=8 and 0<=8.

Reply:
Yeah, that is confusing. I added comments to both the code itself and this test.